docs/api-docs/apps/tutorials/sample-app-next/06-step-4-enhance-ux.md

```diff
@@ -0,0 +1,797 @@
+    const [headerType, setHeaderType] = useState<string>(HeaderTypes.GLOBAL);
+    const router = useRouter();
+    const { pathname } = router;
+
+    useEffect(() => {
+        if (InnerRoutes.includes(pathname)) {
+            // Use InnerHeader if route matches inner routes
+            setHeaderType(HeaderTypes.INNER);
+        } else {
+            // Check if new route matches TabRoutes
+            const tabKey = Object.keys(TabRoutes).find(key => TabRoutes[key] === pathname);
+
+            // Set the active tab to tabKey or set no active tab if route doesn't match (404)
+            setActiveTab(tabKey ?? '');
+            setHeaderType(HeaderTypes.GLOBAL);
+        }
+
+    }, [pathname]);
+
+    useEffect(() => {
+        // Prefetch products page to reduce latency (doesn't prefetch in dev)
+        router.prefetch('/products');
+    });
+
+    const items = [
+        { id: TabIds.HOME, title: 'Home' },
+        { id: TabIds.PRODUCTS, title: 'Products' },
+    ];
+
+    const handleTabClick = (tabId: string) => {
+        setActiveTab(tabId);
+
+        return router.push(TabRoutes[tabId]);
+    };
+
+    if (headerType === HeaderTypes.INNER) return <InnerHeader />;
+
+    return (
+        <Box marginBottom="xxLarge">
+            <Tabs
+                activeTab={activeTab}
+                items={items}
+                onTabClick={handleTabClick}
+            />
+        </Box>
+    );
+};
+
+export default Header;
+```
+
+[View code in GitHub](https://github.com/bigcommerce/sample-app-nodejs/blob/step-4-big-design/components/header.tsx)
+
+The `Header` functional component uses the `useEffect` React hook to perform side effects and enhance performance. Notably, performance enhancement is only visible in a production or production-like environment (integration or staging). `router.prefetch()` will not prefetch the products page while in the development mode.
+
+## Test your app
+
+Your app should now display two tabs: **Home** and **Products**. Click on the **Products** tab. You should see a list of products from your test store.
+
+![Products page](https://storage.googleapis.com/bigcommerce-production-dev-center/images/Sample_app/nextjs-app-14.png "Products page")
+
+### Latency
+
+When loading the products page for the first time, the latency you notice only happens in the development mode. In production or a production-like environment, `router.prefetch()` will prefetch the products page reducing the latency.
+
+To test your app in a production or a production-like environment (integration or staging), run `npm run build` instead of `npm run dev` in your terminal. Doing so will build and compile your local code. Then, run `npm run start` to get the performance enhancements traditionally not available in the development mode. 
+
+Keep in mind that any changes you make to your code after running `npm run build` will need to go through the build process to be captured.
+
+## Create the ErrorMessage component
```

Comment:
we might want to add some additional copy like, "To surface error messages to the app's users, add an error message component. This component can be called from any `page` or `component`."